service/awsconfig/v5/resource/cloudformation/update.go

```diff
@@ -64,23 +65,27 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 		return cloudformation.CreateStackInput{}, microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "debug", "finding out if the main stack should be updated")
-
 	updateState := cloudformation.UpdateStackInput{}
 
-	if currentStackState.Name != "" && !reflect.DeepEqual(desiredStackState, currentStackState) {
-		r.logger.LogCtx(ctx, "debug", "main stack should be updated")
-
-		mainTemplate, err := r.getMainGuestTemplateBody(customObject)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-		updateState.StackName = aws.String(desiredStackState.Name)
-		updateState.TemplateBody = aws.String(mainTemplate)
-		// CAPABILITY_NAMED_IAM is required for updating IAM roles (worker policy).
-		updateState.Capabilities = []*string{
-			aws.String(namedIAMCapability),
+	if updateallowedcontext.IsUpdateAllowed(ctx) {
+		r.logger.LogCtx(ctx, "debug", "finding out if the main stack has to be updated")
+
+		if currentStackState.Name != "" && !reflect.DeepEqual(desiredStackState, currentStackState) {
```

Comment:
So this here is still wrong and I need some feedback. In KVM we can do it that way because scaling is sorted differently and only updates are done during the update process we manage here. FYI in KVM scaling is done via creation/deletion because we manage multiple k8s deployments. In AWS we manage scaling as we manage updates. My question is which properties of the stack state we need to check for identifying a scaling procedure and which properties we need to consider for the update procedure. My assumption would be that we only have to consider the worker count for scaling and all other information for updates. Am I right with this assumption?

Reply:
Yes the WorkerCount is the output that controls scaling. The AWS logic does need to be different because the scaling happens via a stack update.

Reply:
Ok cool. Thanks. I will change the code a bit to sort scaling and updates.